### DISCUSSION TITLE
Problem with custom @request.query in API rules

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#2474] [Q&A] [answered]
I can get all results, but not if I add just the List/Search filter `@request.query.adminView = 1`.  I'm using the filter for a View Collection.  Could there be a problem with views?

This is calling code:

```javascript
  const {items} = await sitesViewCol().getList(page, perPage, {
    $autoCancel: false,
    adminView: 1,
  })
```

Items will be empty if I add the filter above, and it will return 2 records if I make the filter empty.

This is the path `/api/collections/sites_view/records?page=1&perPage=100&adminView=1`

**Comment by ganigeorgiev** [comment, marked as answer]
I think this is because query param values are always string (I'll consider mentioning it in the docs), aka. try with:

```js
@request.query.adminView = "1"
```

**Reply by dschissler** [reply]
Ahh, I just found this.  I was able to get it working by sending and checking for 'abc'.  I saw that you're using that casting library on the Go side and so I figured that it was converting that somehow.

Hey though, maybe you could disallow non-string literals for `@request.query` to prevent others from htting this hard crash?

**Reply by ganigeorgiev** [reply]
> maybe you could disallow non-string literals for @request.query to prevent others from htting this hard crash?

I'll consider it but I don't think we can throw an error since the behavior will be ambiguous because it may not always be possible to resolve upfront the expression operands (_in the above is easy since it is a raw literal, but the operand could be a root field, nested relation field, `@request.data.*` field, etc._)

**Reply by dschissler** [reply]
Perhaps you could just check hard-coded literals and not variables like `@request.data`.

**Reply by dschissler** [reply]
You could disallow it (non-string literals) in the UI and ignore it on the server.
